pkg/stanza/fileconsumer/config.go

```diff
@@ -57,11 +58,12 @@ type Config struct {
 	FingerprintSize         helper.ByteSize       `mapstructure:"fingerprint_size,omitempty"`
 	MaxLogSize              helper.ByteSize       `mapstructure:"max_log_size,omitempty"`
 	MaxConcurrentFiles      int                   `mapstructure:"max_concurrent_files,omitempty"`
-	Splitter                helper.SplitterConfig `mapstructure:",squash,omitempty"`
+	EncodingConfig          helper.EncodingConfig `mapstructure:",squash,omitempty"`
+	Flusher                 helper.FlusherConfig  `mapstructure:",squash,omitempty"`
 }
 
 // Build will build a file input operator from the supplied configuration
-func (c Config) Build(logger *zap.SugaredLogger, emit EmitFunc) (*Manager, error) {
+func (c Config) Build(logger *zap.SugaredLogger, emit EmitFunc, opts ...FactoryOption) (*Manager, error) {
```

Comment:
Thanks for taking the time to review this code. 
In fact, the main purpose of this code is to remove the parameter MultilineConfig from the package `fileconsumer` as you mentioned.
Of course, we can also directly pass in the splitFunc parameter. We also need to consider what to do when the user uses the `MultilineConfig` parameter and passes in a custom `splitFunc` at the same time.

Reply:
> In fact, the main purpose of this code is to remove the parameter MultilineConfig from the package fileconsumer as you mentioned.

Sure, I just want to make sure we are solving the problem you initially proposed in #14593. 

> We also need to consider what to do when the user uses the MultilineConfig parameter and passes in a custom splitFunc at the same time.

I don't know that we need to worry about this case. I think if it is desired to user MultilineConfig, then that is used to generate the SplitFunc in some way. Then passing in the SplitFunc works regardless of whether or not Multiline is used.

Reply:
When the user passes in the SplitFunc, whenever the MultilineConfig is set, the SplitFunc passed in by the user is used. Will this confuse users? Is it possible to consider throwing an error?

Regarding whether to remove from the `fileconsumer` package, we can open an issue for discussion later. And all we need to do now is to go back to the original purpose and let `fileconsumer` support custom SplitFunc.

What do you think about this?

Reply:
> When the user passes in the SplitFunc, whenever the MultilineConfig is set, the SplitFunc passed in by the user is used. Will this confuse users? Is it possible to consider throwing an error?

Sorry, but this I don't think it should work this way, even temporarily. This would be confusing. We should use MultilineConfig to generate the splitFunc, then pass it in.

Reply:
I think the case here is that we did not remove the multilineConfig from the `fileconsumer`, while allowing the user to pass in a custom SplitFunc when calling the Build method.

Reply:
I reconsider your opinion, the `Build` method passes in an Options on the spliterFactory that the responsibility is not clear.
If we add a new method `BuildWithSplitter`, this method can directly pass in a user-defined `splitFunc`. What do you think of this way? @djaglowski

Reply:
This makes more sense to me. I think we should try it. Thanks for finding another solution.

Reply:
This will also not affect the existing interface. I will open a new PR to implement this.